seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java

```diff
@@ -66,13 +69,42 @@ public class KafkaSourceSplitEnumerator implements SourceSplitEnumerator<KafkaSo
         this(metadata, context);
     }
 
+    KafkaSourceSplitEnumerator(ConsumerMetadata metadata, Context<KafkaSourceSplit> context,
+                               long discoveryIntervalMillis) {
+        this(metadata, context);
+        this.discoveryIntervalMillis = discoveryIntervalMillis;
+    }
+
     @Override
     public void open() {
         this.adminClient = initAdminClient(this.metadata.getProperties());
+        ParallelSource parallelSource = context.getParallelSource();
```

Comment:
It's not right the connector to use translation module class. You can use `ScheduledThreadPoolExecutor` to execute topic find with fixed rate. Then close `ScheduledThreadPoolExecutor` when `KafkaSourceSplitEnumerator`  closed.

Reply:
@Hisoka-X Well, I have this idea. I think that if the search thread fails, the consumer thread can be ended. If the search thread fails, do we need to affect the consumer thread

Reply:
It should be, if not, user will think program running fine.

Reply:
The problem I'm facing now is that if I want to stop the source, it seems that I can only interrupt at pollNext，Because I can't get too much valid information in context，Do you have any other good suggestions，